packages/botonic-plugin-contentful/src/cms/contents.ts

```diff
@@ -428,6 +428,58 @@ export class ScheduleContent extends TopContent {
   }
 }
 
+export class OnFinishPayload {
+  readonly type = 'PAYLOAD'
+  constructor(readonly payload: string) {}
+}
+
+export class OnFinishPath {
+  readonly type = 'PATH'
+  constructor(readonly path: string) {}
+}
+
+export type OnFinish = Callback
+
+export class HandoffAgentEmail {
+  readonly type = 'AGENT_EMAIL'
+  constructor(readonly agentEmail?: string) {}
+}
+
+export class HandoffAgentId {
+  readonly type = 'AGENT_ID'
+  constructor(readonly agentId?: string) {}
+}
+
+export class HandoffQueue {
+  readonly type = 'QUEUE'
+  /**
+   * @param queue the queue id or name
+   */
+  constructor(readonly queue?: string) {}
+}
+
+export type HandoffDestination =
+  | HandoffAgentEmail
+  | HandoffAgentId
+  | HandoffQueue
+
+/**
+ * Most CommonFields make no sense for Handoff.
+ * However, we decided to make it a TopContent since it does not depend on other content.
+ * Also CommonFields might be potentially useful.
+ */
+export class Handoff extends TopContent {
+  constructor(
+    readonly common: CommonFields,
+    readonly text: string,
+    readonly onFinish?: OnFinish,
+    readonly destination?: HandoffDestination,
```

Comment:
Add a comment in the code explaining that destination is optional because often it is set dynamically by the bot.
Add a cloneWithDestination method so that the bot can set it dynamically